addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,327 @@
+import gnu.io.SerialPort;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * The {@link RotelRa1xHandler} handles commands and state changes, and communicates
+ * with the amplifier over the serial port interface..
+ *
+ * @author Marius Bjørnstad - Initial contribution
+ */
+public class RotelRa1xHandler extends BaseThingHandler implements Runnable {
+
+    private final static int BAUD = 115200;
+    private int max_vol = 0;
+    private RXTXPort serialPort;
+
+    private boolean connected, exit = false;
+    private volatile boolean power = false;
+
+    private Logger logger = LoggerFactory.getLogger(RotelRa1xHandler.class);
+
+    public RotelRa1xHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        max_vol = ((BigDecimal) getThing().getConfiguration().get("max-vol")).intValue();
+        try {
+            connect();
+        } catch (IOException e) {
+            logger.error("Unable to connect to device", e);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        exit = true;
+        disconnect();
+    }
+
+    private void connect() throws IOException {
+        if (!connected) {
+            if (serialPort != null) {
+                disconnect();
+            }
+            String portName = (String) getThing().getConfiguration().get("port");
+            if (portName == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                throw new IOException("Serial port not configured");
+            } else {
+                try {
+                    serialPort = new RXTXPort(portName);
+                    serialPort.setSerialPortParams(BAUD, SerialPort.DATABITS_8, SerialPort.STOPBITS_1,
+                            SerialPort.PARITY_NONE);
+                } catch (PortInUseException | UnsupportedCommOperationException e) {
+                    serialPort = null;
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    throw new IOException(e);
+                }
+            }
+            connected = true;
+            try {
+                // Don't need continuous updates of the display, we still get updates when
+                // the volume, etc., changes
+                serialPort.getOutputStream().write("display_update_manual!".getBytes("ascii"));
+                updateStatus(ThingStatus.ONLINE);
+                sendForce("get_current_power!");
+                updateState(getThing().getChannel("mute").getUID(), OnOffType.OFF);
+                updateState(getThing().getChannel("dimmer").getUID(), new PercentType(100));
+                Thread receiver = new Thread(this);
+                receiver.start();
+            } catch (IOException e) {
+                disconnect();
+                throw e;
+            }
+        }
+    }
+
+    private void disconnect() {
+        if (connected && serialPort != null) {
```

Comment:
The RXTXPort doesn't provide any method to check if it's closed, and throws a IllegalArgumentException when attempting to close a closed port. http://www.docjava.com/book/cgij/jdoc/gnu/io/RXTXPort.html#close() So I'd rather keep track of whether it's open using the connected flag. I've made some improvements in connect() so that actually works as intended.